big-int/src/operators/multiply.cpp

```diff
@@ -27,10 +27,97 @@
 
 #include <istream>
 #include <ostream>
-
-#include "../bigint.hpp"
+#include <math.h>
+#include <bigint.hpp>
+#include <algorithm>
 
 namespace libbig
 {
+    largeInt largeInt::operator*(const largeInt &next_number) {
+
+        auto append_zeroes = [](const largeInt &x, const int factor) {
+            int k;
+            largeInt Answer = x;
+            for(k = 0; k < factor; k++) {
+                Answer.number.push_back('0');
+            }
+
+            return Answer;
+        };
+
+        auto simple_multiplication = [](const largeInt &x, const largeInt &y) {
+            largeInt x1 = x;
+            largeInt x2 = y;
+            largeInt adder;
+            largeInt Answer(0);
+
+            if(x1.number.length() == 0 || x2.number.length() == 0) {
+                return largeInt();
+            }
+
+            int carry { 0 }, temp, f=1, k;
+            std::string::reverse_iterator i, j;
+            for(i=x1.number.rbegin(); i!=x1.number.rend(); ++i) {
+                for(j=x2.number.rbegin(); j!=x2.number.rend(); ++j) {
+                    temp = (*i - 48) * (*j - 48) + carry;
```

Comment:
alternatively define a helper lambda for this purpose, this will ensure better readability of code.
```C++
auto intConverter = [] (const char& x) {
    return x - '0';
};
```